schedulingApp/unit-tests/test_courses.py

```diff
@@ -1,38 +1,126 @@
+        self.setUp()
+        self.theCourse.setTitle("CS361")
+        self.assertEqual(self.theCourse.getTitle(), "CS361",
+                         msg="Changing title to the currently set title should cause no change to the course")
+    #testing semester changes
+    def test_blankSemester(self):
+        self.setUp()
+        self.assertRaises(TypeError, self.theCourse.setSemester(),
+                          msg="Calling setSemester with no argument should throw TypeError")
 
-    def test_conflictingTitle(self):
-        pass
+    def test_invalidSemester(self):
+        self.setUp()
+        self.assertRaises(ValueError, self.theCourse.setSemester(""),
+                          msg="Changing semester name to a blank string should throw ValueError")
 
-    def test_noChange(self):
-        pass
+    def test_semesterChange(self):
+        self.setUp()
+        self.theCourse.setSemester("Spring 2023")
+        self.assertEqual(self.theCourse.getSemester(), "Spring 2023",
+                         msg="Changing to a valid new semester not successful")
+
+    def test_semesterNoChange(self):
+        self.setUp()
+        self.theCourse.setSemester("Fall 2022")
+        self.assertEqual(self.theCourse.getSemester(), "Fall 2022",
+                         msg="Changing semester to the currently set semester should cause no change to the course")
+    #general course change tests
+    def test_conflictingCourse(self):
+        self.setUp()
+        a = Course("CS423", "Fall 2022")
+        a.title = "CS423"
+        a.semester = "Fall 2022"
+        a.setTitle("CS361")
+        self.assertFalse(a.title == "CS361",
+                         msg="Should not allow multiple courses with the same semester and title")
+
+    def test_sameNameDifferentSemester(self):
+        self.setUp()
+        a = Course("CS423", "Spring 2023")
+        a.title = "CS423"
+        a.semester = "Spring 2023"
+        a.setTitle("CS361")
+        self.assertEqual(a.title, "CS361",
+                         msg="Should update to the same name as a different course as long as semesters are different.")
 
-    def test_validEdit(self):
-        pass
 
 
 class DeleteCourse(TestCase):
+    theCourse = None
+    def setUp(self):
+        theCourse = Course("CS361", "Fall 2022")
+        theCourse.title = "CS361"
+        theCourse.semester = "Fall 2022"
+
     def test_courseNotFound(self):
-        pass
+        self.setUp()
+        self.assertRaises(NameError, self.myCourse.__del__(),
+                          msg="Trying to delete nonexistent course should throw NameError")
 
     def test_validAndExistingCourse(self):
-        pass
+        self.setUp()
+        self.theCourse.__del__()
+        self.assertFalse(self.theCourse.getTitle() == "CS361", msg="Title for deleted course should no longer exist.")
+        self.assertFalse(self.theCourse.getSemester() == "Fall 2022",
+                         msg="Semester for deleted course should no longer exist.")
```

Comment:
Not sure about this one, but if delete is called, shouldnt getTitle and getSemester error, ill bring this up at the meeting tomorrow to be sure

Reply:
Updated deletion testing, should look better now.